### ISSUE TITLE
There's already a module with namespace

### ISSUE THREAD

**Issue by amk221** [sass/sass#2778]
The Ember.js application I work on is structured in 'pods'.

This means that the stylesheet is co-located with the associated template and javascript.

For example

```
/my-component
     /style.scss
     /component.js
     /template.hbs
```

When I recently attempted to migrate from `@import` to `@use` I get the error

> There's already a module with namespace "style"

**Comment by jathak** [comment]
That error occurs when attempting to `@use` two modules with the same namespace. The default namespace is the basename of the file passed to the `@use` rule, so if you had something like:

```
@use "component-a/style";
@use "component-b/style";
```

you would get this error because both of those rules use the namespace `style`. To fix this manually, you can use an `as` clause to manually specify the namespace for a `@use` rule. e.g.

```
@use "component-a/style" as a-style;
@use "component-b/style" as b-style;
```

To make the migration easier, you can use the [Sass migrator](https://sass-lang.com/documentation/cli/migrator#module), which should automatically add `as` clauses when necessary (it will also automatically add these namespaces to any variable, mixin, or function references from other files, so you don't have to do that manually either). If you already used the migrator and the stylesheets it output resulted in this error, then that's a bug with it, and you should file an issue [here](https://github.com/sass/migrator/issues/new).

**Comment by nex3** [comment]
@amk221 if you still need help, can you post the stylesheet in question here so we can help debug it?

**Comment by amk221** [comment]
Thanks for the quick response. I should have elaborated a bit more. 

Essentially, coming from ES6 modules, I just had the wrong expectations of SCSS modules. Take this contrived example:

```css
/* variables.scss */
$red: '#f00';
```
```css
/* mixins.scss */
@use 'variables' as vars;

@mixin red {
  color: vars.$red;
}
```
```css
/* my-component/style.scss */

@use 'mixins' as *;

.my-component {
  @include red;
}
```
```css
/* my-other-component/style.scss */

@use 'mixins' as *;

.my-other-component {
  @include red;
}
```


So far, the module system makes sense. Because there is no actual output.

The bit that tripped me up was the output stage...

I changed:

```css
/* app.scss */
@import 'my-component/style';
@import 'my-other-component/style';
```

To this:

```css
/* app.scss */
@use 'my-component/style';
@use 'my-other-component/style';
```

...and expected it to work.

I assumed I only needed `as` if I was going to _refer_ to the members. Which I never do - I'm just relying on `use` to build the output.

So it's a bit long winded, but the solution of course is:


```css
/* app.scss */
@use 'my-component/style' as my-component;
@use 'my-other-component/style' as my-other-component;
```

The TLDR is user error.

But... If the module was unique by file path rather than file name, this problem would go away?

**Comment by amk221** [comment]
After working with SASS modules for a while now. I can better articulate what I want.

And that is, `@use` to never output anything

And something like `@output 'my-file'`

Therefore, there would never be any reason to specify a unique name when outputting.

**Comment by amk221** [comment]
I don't consider my original issue and issue, but, I've just re-opened it to re-start the conversation.

**Comment by nex3** [comment]
You can make this happen yourself by only ever writing mixins in partials, and using `@include` to actually generate the CSS in your entrypoint file. This is a common style at Google, and we loosely recommend it for most larger applications.

Given that there's a userland way of making this work, I don't think it makes sense to add new syntax for it.

**Comment by amk221** [comment]
Perhaps I didn't explain myself properly.

Imagine:

```css
/* components/foo/style.scss */
@use 'something';
.foo { @include something.example; }
```

```css
/* components/bar/style.scss */
@use 'something-else';
.bar { @include something-else.example; }
```

...then, seeing as though sass.org says
> "The Sass team discourages the continued use of the @import rule"

we ought to change this:

```css
/* app.scss */
@import 'foo/style';
@import 'bar/style';
```

Into this:

```css
/* app.scss */
@use 'foo/style';
@use 'bar/style';
```

But, that _needlessly_ requires us to name the `@use`'s, like:

```css
/* app.scss */
@use 'foo/style' as foo;
@use 'bar/style' as bar;
```

At the output stage this is _entirely pointless_. Hence my suggestion for `@output`

```css
/* app.scss */
@output 'foo/style';
@output 'bar/style';
```

**Comment by amk221** [comment]
Thereby, `@use` being specifically for utilising _modules_ and `@output` being responsible for outputting. Two separate concerns.

**Comment by mirisuzanne** [comment]
That [functionality already exists](https://sass-lang.com/documentation/modules/meta#load-css) with the `meta.load-css($url, $with)` Mixin.

```scss
@use 'sass:meta';

/* app.scss */
@include meta.load-css('foo/style');
@include meta.load-css('bar/style');
```

**Comment by nex3** [comment]
Note that `meta.load-css()` is substantially slower and somewhat more error-prone than a plain `@use` rule.

@amk221 It's true that naming all your files `_style.scss` is awkward with the new module system, especially if you don't want to use any members from them. I recommend naming them `_index.scss` instead so you can just write:

```scss
@use "foo";
@use "bar";
```

**Comment by amk221** [comment]
Thanks @nex3, that helps a little, apart from this use case:

```css
@use 'routes/people/index/index' as people-index;
@use 'routes/people/show/index' as people-show;
@use 'routes/people/create/index' as people-create;
@use 'routes/tags/index/index' as tags-index;
@use 'routes/tags/show/index' as tags-show;
@use 'routes/tags/create/index' as tags-create;
```

...but I can probably live with that. 

In case it's not clear, the above is required because this doesn't work for the two reasons mentioned in the comments:

```css
@use 'routes/people/index'   /* "routes/people/index" doesn't automatically pick up "index.scss" inside the index directory */
@use 'routes/people/show';
@use 'routes/people/create';
@use 'routes/tags/index'; 
@use 'routes/tags/show';    /* "show" clashes with the people/show, etc... */
@use 'routes/tags/create';
```
